network/network_windows_test.go

```diff
@@ -400,3 +401,32 @@ func TestAddIPv6DefaultRouteUnhappyPathAddRoute(t *testing.T) {
 		t.Fatal("Failed to test unhappy path with failing to add default route command")
 	}
 }
+
+func TestNewNetworkImplHnsV2ForBackendNIC(t *testing.T) {
+	pnpID := "PCI\\VEN_15B3&DEV_101C&SUBSYS_000715B3&REV_00\\5&8c5acce&0&0"
+
+	nm := &networkManager{
+		ExternalInterfaces: map[string]*externalInterface{},
+		plClient:           platform.NewMockExecClient(false),
+	}
+
+	nwInfo := &EndpointInfo{
+		NetworkID:    "d3f97a83-ba4c-45d5-ba88-dc56757ece28",
+		MasterIfName: "ib1",
+		Mode:         "transparent",
+		NICType:      cns.BackendNIC,
+		PnPID:        pnpID,
+	}
+
+	extInterface := &externalInterface{
+		Name: "eth1",
+	}
+
+	Hnsv2 = hnswrapper.NewHnsv2wrapperFake()
+
+	// should return nil if nicType is BackendNIC when creating network
+	network, err := nm.newNetworkImplHnsV2(nwInfo, extInterface)
```

Comment:
lets validate len(networks) in hsv2 cache to make sure hns network create api not invoked. we can also write one similar test for endpoint creation

Reply:
we skipped the network creation as discussed(nm.CreateNetwork); so we won't hit the network creation since beginning; once we figure out how to get ib interface name from windows team, we can add this test case.

Reply:
Since we discussed we skipped the network creation from the beginning and it won't hit network creation at all; let's add this ut when we figure out the ib interface name from windows team. what do you think?

Reply:
will add this in the next PR

Reply:
hope you address this in another PR